src/basics.php

```diff
@@ -143,3 +143,20 @@ function json_last_error_msg()
     }
 
 }
+
+if (!function_exists('breakpoint')) {
+    /**
+     * Command to return the eval-able code to startup PsySH in interactive debugger
+     * Works the same way as eval(\Psy\sh());
+     * ```
+     * eval(breakpoint());
+     * ```
+     * @return string
+     */
+    function breakpoint()
+    {
+        if (PHP_SAPI === 'cli' && class_exists('\Psy\Shell')) {
+            return 'extract(\Psy\Shell::debug(get_defined_vars(), isset($this) ? $this : null));';
```

Comment:
You might be able to use debug_backtrace() to walk up the stack frames and get the defined variables in the calling context. That might make it possible to not need the eval() in the caller.